src/gui/executionlogwidget.cpp

```diff
@@ -29,78 +29,73 @@
 #include "executionlogwidget.h"
 
 #include <QDateTime>
+#include <QMenu>
 #include <QPalette>
 
 #include "base/global.h"
-#include "loglistwidget.h"
+#include "base/logger.h"
+#include "log/logfiltermodel.h"
+#include "log/loglistview.h"
+#include "log/logmodel.h"
 #include "ui_executionlogwidget.h"
 #include "uithememanager.h"
 
-ExecutionLogWidget::ExecutionLogWidget(QWidget *parent, const Log::MsgTypes &types)
+ExecutionLogWidget::ExecutionLogWidget(QWidget *parent, const Log::MsgTypes types)
     : QWidget(parent)
     , m_ui(new Ui::ExecutionLogWidget)
-    , m_msgList(new LogListWidget(MAX_LOG_MESSAGES, types, this))
-    , m_peerList(new LogListWidget(MAX_LOG_MESSAGES, Log::ALL, this))
+    , m_messageModel(new LogMessageModel(this))
+    , m_peerModel(new LogPeerModel(this))
+    , m_messageFilterModel(new LogFilterModel(types, this))
+    , m_messageView(new LogListView(this))
+    , m_peerView(new LogListView(this))
 {
     m_ui->setupUi(this);
 
+    m_messageFilterModel->setSourceModel(m_messageModel);
+    m_messageView->setModel(m_messageFilterModel);
+    m_messageView->setContextMenuPolicy(Qt::CustomContextMenu);
+    connect(m_messageView, &LogListView::customContextMenuRequested, this, [this](const QPoint &pos) {
+        displayContextMenu(pos, m_messageView, m_messageModel);
+    });
+
+    m_peerView->setModel(m_peerModel);
+    m_peerView->setContextMenuPolicy(Qt::CustomContextMenu);
+    connect(m_peerView, &LogListView::customContextMenuRequested, this, [this](const QPoint &pos) {
```

Comment:
Fix coding style.

Reply:
I think I need to put the braces in the new line.
done.